xgcm/grid.py

```diff
@@ -861,7 +861,26 @@ def transform(
                 "`transform` can only be used on axes that are non-periodic. Pass `periodic=False` to `xgcm.Grid`."
             )
 
-        def _parse_target(target, target_dim, target_data_dim, target_data):
+        def _target_data_name_handling(target_data):
+            """Handle target_data input without a name"""
+            if target_data.name is None:
+                warnings.warn(
+                    "The input dataarray `target_data` has no name. Will default to `_UNNAMED_`."
+                )
+                da.name = "_UNNAMED_"
+
+        def _check_target_alignment(da, target_da):
```

Comment:
```suggestion
        def _check_target_alignment(target_da):
```

to be consistent with the other nested functions.

Reply:
I dont think this works. We need `da` in `_check_target_alignment`.

Reply:
nested functions inherit all variables in the parent function's scope AFAIk (also you're using `da` in `_target_data_name_handling` without it being in the signature)

Reply:
Oh damn. Pardon my ignorance. I'll change